CONTRIBUTING.md

```diff
@@ -45,31 +45,64 @@ Feature requests are welcome, but take a moment to find out whether your idea fi
 
-Plugins which fall under the following categories will not be implemented or considered and the request will be closed:
+Information about the website, as well as explicit URLs from your web browser's URL address bar are required, as are details regarding the website, what it is, who it is run by, and more. This allows for easier plugin review and implementation. We reserve the right to refuse to implement, develop, or update any plugin. In addition, we may remove existing plugins at our own discretion at any time without reasoning.
 
-1. Any kind of streaming service that uses DRM protection
+Before you request a plugin, please be aware that this will merely be a suggestion for the project maintainers (or anyone else) to have a look at the website and implement a new plugin for it. In the vast majority of cases, no one except you will initially have an interest in such a plugin, so you should think about whether it will be useful for other users of Streamlink.
 
-2. Sites which are hosting stolen content as their main source of content
+Implementing new plugins can be a lot of effort, as well as keeping them maintained. Providing plugin code in a pull request written on your own after it has been discussed and approved in the request issue is therefore much appreciated, if the code quality is acceptable. Further assistance in plugin maintenance via plugin issues or bugfix pull requests will be greatly appreciated as well.
 
-3. Sites which are primarily rehosting content that is available from a legitimate source (TV shows, sports, news, etc.)
+Please remember that custom plugins can always be [sideloaded][plugin-sideloading], in case your request will be rejected or if plugins will be removed from Streamlink's `master` branch.
 
-4. Sites which require any sort of cable login or subscription
+**Plugins which fall under the following categories will not be implemented and any requests will be rejected.** Depending on the case, already existing plugins which don't meet these plugin criteria, may be kept.
 
-5. NSFW sites of a pornographic nature (cam sites, porn sites, etc.)
 
-6. Sites which don't provide any real live streaming content, eg. only VODs or VODs being rebroadcasted
+### Plugin rules
 
-7. Sites which don't provide any video streaming content, eg. radio stations
+1. **Any kind of streaming service that uses DRM protection**
 
-8. Sites where the maintainer has requested we not add their site to Streamlink
+    Streamlink won't implement or break any DRM schemes.
 
-9. Sites which are unmaintained, are in beta or are undergoing heavy amounts of development and may change rapidly
+2. **Sites which are hosting stolen content as their main source of content**
 
-10. Sites which have no way to determine viewership numbers
+    Plugins must only support websites with authentic content.
```

Comment:
How are we defining "authentic" here? This was an annoying spot we got in to in the past when people asked us to implement some weird TV station that was running news broadcasts or other local content. Does this need to be more explicit? I don't think a nightly news broadcast for a town of 20 people really counts as authentic or non-niche.

Reply:
I believe you're confusing two rules here. Niche content is already covered by rule 10.

Reply:
No I'm aware of rule 10 but my assumption is most people aren't going to read all the rules so I'm not sure if we should be more explicit in this one. It's not a major deal or anything, I just don't want the headache of arguing with people about what the word "authentic" means so I figured I'd mention it. We can revise later if it becomes a problem.